tests/h/search/index_test.py

```diff
@@ -48,72 +46,6 @@ def test_it_indexes_presented_annotation(
         )
         assert indexed_ann == presenter.asdict.return_value
 
-    def test_it_can_index_an_annotation_with_no_document(
-        self, factories, index_annotations, get_indexed_ann
-    ):
-        annotation = factories.Annotation.build(document=None)
-
-        index_annotations(annotation)
-
-        assert get_indexed_ann(annotation.id)["document"] == {}
-
-    def test_it_indexes_the_annotations_document_web_uri(
-        self, factories, index_annotations, get_indexed_ann
-    ):
-        annotation = factories.Annotation.build(
-            document=factories.Document.build(
-                web_uri="https://example.com/example_article"
-            )
-        )
-
-        index_annotations(annotation)
-
-        # *Searching* for an annotation by ``annotation.document`` (e.g. by
-        # document ``title`` or ``web_uri``) isn't enabled.  But you can
-        # retrieve an annotation by ID, or by searching on other field(s), and
-        # then access its ``document``. Bouncer
-        # (https://github.com/hypothesis/bouncer) accesses h's Elasticsearch
-        # index directly and uses this ``document`` field.
-        assert (
-            get_indexed_ann(annotation.id)["document"]["web_uri"]
-            == "https://example.com/example_article"
-        )
```

Comment:
It's certainly useful to have a test for this document `web_uri` indexing behaviour that bouncer relies on, but this isn't the right place for it. It's actually a feature of `DocumentSearchIndexPresenter` and is tested in the `DocumentSearchIndexPresenter` unit tests. And the fact that `AnnotationSearchIndexPresenter` uses `DocumentSearchIndexPresenter` is tested in the `AnnotationSearchIndexPresenter` unit tests. `index()`'s job is just to call `AnnotationSearchIndexPresenter.asdict()` and pass the results to Elasticsearch, so that's all that could be tested here.

I think the comment above the `assert` here is useful though. Maybe consider moving it here https://github.com/hypothesis/h/blob/9bc3e502fa6a9d4d2e1f5ba5d4044c14f45a6bab/tests/h/presenters/document_searchindex_test.py#L14-L19

Reply:
I've done this in a tiny other PR so as not to cause rebasing woe: https://github.com/hypothesis/h/pull/6262